keps/sig-storage/1495-volume-populators/README.md

```diff
@@ -271,6 +294,153 @@ downgrade.
 
 No issues
 
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [X] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: AnyVolumeDataSource
+    - Components depending on the feature gate: kube-apiserver
+  - [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+* **Does enabling the feature change any default behavior?**
+  Before this feature, kube API server would silently drop any PVC data source
+  that it didn't recognize, as if the client never populated the field. After
+  enabling this feature, the API server allows any object to be specified.
+  An external admission controller will take over validation of the field,
+  and its response to invalid data sources will be to fail the create operation,
+  rather than silently drop them.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Turning off the feature gate should drop all the data sources that were not
+  already valid and restore the old behavior.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  PVC which previously had data sources that were invalided by the rollback
+  would be empty volumes. Users would need to delete and re-create those PVCs.
+
+* **Are there any tests for feature enablement/disablement?**
+  Not at this time.
```

Comment:
Do you have plan to add these tests before going beta?  If so, please add here.

Reply:
The commends in the template suggested that e2e testes for feature disablement can't work right now. All we will have is unit tests, and those already exist.

Reply:
> The commends in the template suggested that e2e testes for feature disablement can't work right now. All we will have is unit tests, and those already exist.

Can you clarify that you'll have a unit test to ensure that validation with the featuregate off that has the data as it would look if the feature had been enabled and then disabled?  We've had issues in the past where a discriminated union became invalid for instance.


```go
OneOfSomehing struct{
  Type string // either A or B
  A *struct
  B *struct
}
```

When disabled, the validation function gets 
```go
type: A
A: nil
B: nil
```

The update must succeed and a unit test can check this.